util/topgen-fusesoc.py

```diff
@@ -0,0 +1,177 @@
+    # pwrmgr_reg_pkg generated by topgen.
+    if reg_only:
+        for ip in ['alert_handler', 'clkmgr', 'flash_ctrl', 'padctrl', 'pinmux', 'pwrmgr',
+                   'rstmgr', 'rv_plic']:
+            core_filepath = os.path.abspath('generated-%s.core' % ip)
+            name = 'lowrisc:ip:%s_reggen' % ip,
+            files = ['ip/%s/rtl/autogen/%s_reg_pkg.sv' % (ip, ip),
+                     'ip/%s/rtl/autogen/%s_reg_top.sv' % (ip, ip)]
+            generated_core = {
+                'name': '%s' % name,
+                'filesets': {
+                    'files_rtl': {
+                        'depend': [
+                            'lowrisc:ip:tlul',
+                        ],
+                        'files': files,
+                        'file_type': 'systemVerilogSource'
+                    },
+                },
+                'targets': {
+                    'default': {
+                        'filesets': [
+                            'files_rtl',
+                        ],
+                    },
+                },
+            }
+            write_core(core_filepath, generated_core)
+
+    else:
+        core_filepath = os.path.abspath('generated-topgen.core')
+        generated_core = {
+            'name': "lowrisc:systems:generated-topgen",
+            'filesets': {
+                'files_rtl': {
+                    'depend': [
+                        # flash_ctrl
+                        'lowrisc:constants:top_pkg',
+                        'lowrisc:prim:util',
+                        'lowrisc:ip:lc_ctrl_pkg',
+                        'lowrisc:ip:pwrmgr_pkg',
+                        # rstmgr
+                        'lowrisc:prim:clock_mux2',
+                        # clkmgr
+                        'lowrisc:prim:all',
+                        'lowrisc:prim:clock_gating',
+                        'lowrisc:prim:clock_buf',
+                        'lowrisc:prim:clock_div',
+                        # Top
+                        # ast and sensor_ctrl not auto-generated, re-used from top_earlgrey
+                        'lowrisc:systems:sensor_ctrl',
+                        'lowrisc:systems:ast_wrapper_pkg',
+                    ],
+                    'files': [
+                        # IPs
+                        'ip/clkmgr/rtl/autogen/clkmgr_pkg.sv',
+                        'ip/clkmgr/rtl/autogen/clkmgr.sv',
+                        'ip/flash_ctrl/rtl/autogen/flash_ctrl_pkg.sv',
+                        'ip/flash_ctrl/rtl/autogen/flash_ctrl.sv',
+                        'ip/rstmgr/rtl/autogen/rstmgr_pkg.sv',
+                        'ip/rstmgr/rtl/autogen/rstmgr.sv',
+                        'ip/rv_plic/rtl/autogen/rv_plic.sv',
+                        # Top
+                        'rtl/autogen/top_englishbreakfast_rnd_cnst_pkg.sv',
+                        'rtl/autogen/top_englishbreakfast_pkg.sv',
+                        'rtl/autogen/top_englishbreakfast.sv',
```

Comment:
would it make sense to use `topname` to make the filenames parameteric here?

Reply:
Oh, yes absolutely! I will change that.

Reply:
This is done now.